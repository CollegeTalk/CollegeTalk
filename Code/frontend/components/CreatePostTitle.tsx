import {TouchableOpacity, Platform, KeyboardAvoidingView, TextInput, StyleSheet, Text, View } from 'react-native';
import React from 'react';

const CreatePostTitle = () => {
    return (
    <View style = {styles.Center}>
        {/* Write a post title */}
        <KeyboardAvoidingView
                behavior = {Platform.OS == "ios" ? "padding" : "height"}
                style ={styles.writeTaskWrapper} 
            >
                
            <TextInput style ={styles.input} placeholder = {"Write a post title"}/>
        </KeyboardAvoidingView>
    </View>
    );
}

const styles = StyleSheet.create({
    Center: {
        alignItems: 'center',
        justifyContent: 'space-around',
        marginTop: 100
        
    },

    input: {
        paddingVertical: 15,
        paddingHorizontal: 15,
        backgroundColor: '#FFF',
        borderRadius: 60,
        borderColor: '#C0C0C0',
        borderWidth: 1,
        width: 250
    },

    writeTaskWrapper:{
        position: 'absolute',
        bottom: 60,
        width: '100%',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center'
    }
  });

export default CreatePostTitle;