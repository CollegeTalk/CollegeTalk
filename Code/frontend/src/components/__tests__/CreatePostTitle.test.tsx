import renderer from "react-test-renderer";

import CreatePostTitle from "../CreatePost/CreatePostTitle";

describe("<CreatePostTitle />", () => {
    it("has 1 child", () => {
        const tree = renderer
            .create(<CreatePostTitle titleText="" setTitleText={() => null} />)
            .toJSON() as renderer.ReactTestRendererJSON;
        expect(tree.children!.length).toBe(1);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(<CreatePostTitle titleText="" setTitleText={() => null} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
