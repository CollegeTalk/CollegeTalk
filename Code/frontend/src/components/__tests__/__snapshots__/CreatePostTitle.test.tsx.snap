// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreatePostTitle /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "space-around",
      "marginTop": 100,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "bottom": 60,
          "flexDirection": "row",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Write a post title"
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderColor": "#C0C0C0",
          "borderRadius": 60,
          "borderWidth": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 15,
          "width": 250,
        }
      }
      value=""
    />
  </View>
</View>
`;
